src/Control/Comonad/Cofree/Cofreer.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE FlexibleInstances, GADTs, MultiParamTypeClasses, TypeFamilies, UndecidableInstances #-}
+module Control.Comonad.Cofree.Cofreer
+( CofreerF(..)
+, Cofreer(..)
+, headF
+, tailF
+, cowrap
+, coiter
+, unfold
+, extract
+, unwrap
+) where
+
+import Control.Arrow ((&&&))
+import Control.Comonad
+import Control.Comonad.Cofree.Class
+import Data.Bifunctor
+import Data.Functor.Classes
+import Data.Functor.Foldable hiding (unfold)
+
+data CofreerF f a b where
+  Cofree :: a -> (x -> b) -> f x -> CofreerF f a b
+
+headF :: CofreerF f a b -> a
+headF (Cofree a _ _) = a
+
+tailF :: Functor f => CofreerF f a b -> f b
+tailF (Cofree _ t r) = t <$> r
+
+
+newtype Cofreer f a = Cofreer { runCofreer :: CofreerF f a (Cofreer f a) }
+
+infixr 5 `cowrap`
+cowrap :: a -> f (Cofreer f a) -> Cofreer f a
+cowrap a r = Cofreer (Cofree a id r)
+
+coiter :: Functor f => (b -> f b) -> b -> Cofreer f b
+coiter f = unfold (id &&& f)
+
+unfold :: Functor f => (b -> (a, f b)) -> b -> Cofreer f a
+unfold f c = let (x, d) = f c in Cofreer (Cofree x (unfold f) d)
+
+
+-- Instances
+
+instance Bifunctor (CofreerF f) where
+  bimap f g (Cofree a t r) = Cofree (f a) (g . t) r
+
+instance Functor (CofreerF f a) where
+  fmap = second
+
+instance Functor (Cofreer f) where
+  fmap f = Cofreer . bimap f (fmap f) . runCofreer
+
+instance Comonad (Cofreer f) where
+  extract (Cofreer (Cofree a _ _)) = a
+  extend f c@(Cofreer (Cofree _ t r)) = Cofreer (Cofree (f c) (extend f . t) r)
+
+instance Functor f => ComonadCofree f (Cofreer f) where
+  unwrap = tailF . runCofreer
+
+
+instance Foldable f => Foldable (CofreerF f a) where
+  foldMap f (Cofree _ t r) = foldMap (f . t) r
+
+instance Foldable f => Foldable (Cofreer f) where
+  foldMap f (Cofreer c) = mappend (f (headF c)) (foldMap (foldMap f) c)
+
+
+type instance Base (Cofreer f a) = CofreerF f a
+
+instance Recursive (Cofreer f a) where project = runCofreer
+instance Corecursive (Cofreer f a) where embed = Cofreer
```

Comment:
I didn’t want orphan instances, so I wrote this module. In hindsight, that seems a little absurd.